editor/components/theme-support-check/index.js

```diff
@@ -1,16 +1,35 @@
 /**
  * External dependencies
  */
-import { get, some, castArray } from 'lodash';
+import {
+	castArray,
+	includes,
+	isArray,
+	get,
+	some,
+} from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 
-export function ThemeSupportCheck( { themeSupports, children, supportKeys } ) {
+export function ThemeSupportCheck( { themeSupports, children, postType, supportKeys } ) {
 	const isSupported = some(
-		castArray( supportKeys ), ( key ) => get( themeSupports, [ key ], false )
+		castArray( supportKeys ), ( key ) => {
+			const supported = get( themeSupports, [ key ], false );
+			// 'post-thumbnails' can be boolean or an array of post types.
+			// In the latter case, we need to verify `postType` exists
+			// within `supported`. If `postType` isn't passed, then the check
+			// should fail.
+			if ( 'post-thumbnails' === key && isArray( supported ) ) {
+				if ( ! postType ) {
```

Comment:
In what (real-world) conditions do you imagine this condition would be satisfied?

Aside: If we keep it, we don't have a test case for it.

Aside 2: If it were the case that `postType` were undefined / falsey, wouldn't the `includes` logic still be sufficient to return `false` ?